patches/minecraft/net/minecraft/world/WorldProvider.java.patch

```diff
@@ -65,30 +65,34 @@
      }
  
      @SideOnly(Side.CLIENT)
-@@ -241,6 +216,371 @@
+@@ -241,6 +216,375 @@
          return new WorldBorder();
      }
  
 +    /*======================================= Forge Start =========================================*/
 +    private net.minecraftforge.client.IRenderHandler skyRenderer = null;
 +    private net.minecraftforge.client.IRenderHandler cloudRenderer = null;
 +    private net.minecraftforge.client.IRenderHandler weatherRenderer = null;
-+    private int dimensionId;
++    private net.minecraftforge.common.DimensionProvider dimensionProvider;
 +
 +    /**
-+     * Sets the providers current dimension ID, used in default getSaveFolder()
++     * Sets the providers current dimensionProvider ID, used in default getSaveFolder()
 +     * Added to allow default providers to be registered for multiple dimensions.
-+     * This is to denote the exact dimension ID opposed to the 'type' in WorldType
++     * This is to denote the exact dimensionProvider ID opposed to the 'type' in WorldType
 +     *
-+     * @param dim Dimension ID
++     * @param dimID DimensionProvider ID
 +     */
-+    public void setDimension(int dim)
++    public void setDimensionProvider(net.minecraft.util.ResourceLocation dimID)
 +    {
-+        this.dimensionId = dim;
++        this.dimensionProvider =net.minecraftforge.fml.common.registry.ForgeRegistries.DIMENSIONS.getValue(dimID);
 +    }
-+    public int getDimension()
++    public net.minecraft.util.ResourceLocation getDimensionProvider()
 +    {
-+        return this.dimensionId;
++        return dimensionProvider.getID();
++    }
++    public int getDimensionProviderInt()
```

Comment:
This is a poor name, perhaps `getDimensionId()`?

Reply:
My point is more that it isn't the ID of the provider, but the underlying dimension.

Reply:
Yes, so changing it to `getDimensionID()` would not make a lot of sense when `WorldProvider` is renamed? This will definitely only get merged in 1.13 so I'm basing names around the MCP mappings there.